pkg/cloudprovider/providers/gce/gce_instances.go

```diff
@@ -157,7 +156,20 @@ func (gce *GCECloud) ExternalID(nodeName types.NodeName) (string, error) {
 // InstanceExistsByProviderID returns true if the instance with the given provider id still exists and is running.
 // If false is returned with no error, the instance will be immediately deleted by the cloud controller manager.
 func (gce *GCECloud) InstanceExistsByProviderID(providerID string) (bool, error) {
-	return false, errors.New("unimplemented")
+	project, zone, name, err := splitProviderID(providerID)
+	if err != nil {
+		return false, err
+	}
+
+	_, err = gce.getInstanceFromProjectInZoneByName(project, zone, name)
```

Comment:
Can we create an instanceByProviderID method? Then both InstanceExistsByProviderID and InstanceTypeByProviderID can both call that.

Reply:
Ok, done.